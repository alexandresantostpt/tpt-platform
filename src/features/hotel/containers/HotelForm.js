import { connect } from 'react-redux'

import { requestDelAction, requestGetSingleAction, requestSaveDataAction } from '../actions'
import { requestGetSingleAction as travelRequestGetSingleAction } from '@features/travelScript/actions'

import { getObjSelector } from '../selectors'
import { getObjSelector as travelGetObjSelector, getClientsSelector } from '@features/travelScript/selectors'
import HotelForm from '../pages/HotelForm'

import { resetSuggestionsAction, searchLibraryAction } from '@features/library/actions'
import { getListSelector } from '@features/library/selectors'

const mapStateToProps = state => ({
    clients: getClientsSelector(state),
    hotel: getObjSelector(state),
    hotelSuggestions: getListSelector(state),
    travelScript: travelGetObjSelector(state)
})

const mapDispatchToProps = {
    requestDelAction,
    requestGetSingleAction,
    requestSaveDataAction,
    resetSuggestionsAction,
    searchLibraryAction,
    travelRequestGetSingleAction
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(HotelForm)
