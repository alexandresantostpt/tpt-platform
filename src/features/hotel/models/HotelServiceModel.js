import { config } from '@config'
import moment from 'moment'

import { normalizeTimezone } from '@utils/date'

import HotelLibraryModel from './HotelLibraryModel'

class HotelServiceModel {
    constructor(
        {
            __v,
            _id,
            checkInDate,
            checkInHour,
            checkOutDate,
            checkOutHour,
            passengers,
            createdAt,
            guestCount,
            library,
            reserveCode,
            roomType,
            scriptDate,
            updatedAt
        },
        fromForm = false
    ) {
        this.__v = __v
        this._id = _id
        this._checkInDate = moment(checkInDate).utc(false)
        this._checkInHour = fromForm ? normalizeTimezone(checkInHour) : moment(checkInHour).utc(false)
        this._checkOutDate = moment(checkOutDate).utc(false)
        this._checkOutHour = fromForm ? normalizeTimezone(checkOutHour) : moment(checkOutHour).utc(false)
        this._passengers = passengers
        this._createdAt = moment(createdAt).utc(false)
        this._guestCount = guestCount
        this._library = library ? new HotelLibraryModel(library, fromForm) : null
        this._reserveCode = reserveCode
        this._roomType = roomType
        this._scriptDate = scriptDate
        this._updatedAt = moment(updatedAt).utc(false)
    }

    get v() {
        return this.__v
    }

    get id() {
        return this._id
    }

    get checkInDate() {
        return this._checkInDate
    }

    get checkInHour() {
        return this._checkInHour
    }

    get checkOutDate() {
        return this._checkOutDate
    }

    get checkOutHour() {
        return this._checkOutHour
    }

    get passengers() {
        return this._passengers
    }

    get createdAt() {
        return this._createdAt
    }

    get guestCount() {
        return this._guestCount
    }

    get library() {
        return this._library
    }

    get reserveCode() {
        return this._reserveCode
    }

    get roomType() {
        return this._roomType
    }

    get scriptDate() {
        return this._scriptDate
    }

    get updatedAt() {
        return this._updatedAt
    }

    getImage() {
        return this.library && this.library.image && `${config.api.url}/hotel/download/${this.library.image}`
    }

    /* eslint-disable-next-line complexity */
    toDTO() {
        return {
            __v: this.v,
            _id: this.id,
            checkInDate: this.checkInDate || null,
            checkInHour: this.checkInHour || null,
            checkOutDate: this.checkOutDate || null,
            checkOutHour: this.checkOutHour || null,
            guestCount: this.guestCount || null,
            library: this.library ? this.library.toDTO() : null,
            passengers: this.passengers || null,
            reserveCode: this.reserveCode || null,
            roomType: this.roomType || null,
            scriptDate: this.scriptDate || null
        }
    }

    toForm() {
        return {
            __v: this.v,
            _id: this.id,
            checkInDate: this.checkInDate,
            checkInHour: this.checkInHour,
            checkOutDate: this.checkOutDate,
            checkOutHour: this.checkOutHour,
            guestCount: this.guestCount,
            library: this.library ? this.library.toForm() : null,
            passengers: this.passengers,
            reserveCode: this.reserveCode,
            roomType: this.roomType,
            scriptDate: this.scriptDate
        }
    }
}

export default HotelServiceModel
