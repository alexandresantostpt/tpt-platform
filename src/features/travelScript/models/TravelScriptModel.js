import moment from 'moment'

import Travel from '../../travel/TravelModel'
import TravelDateModel from './TravelDateModel'
import { isNotEmpty } from '@/utils/functions'

class TravelScriptModel {
    constructor({ __v, _id, dates, deleted, edited, travel, createdAt, updatedAt }) {
        this.__v = __v
        this._id = _id
        this._dates = dates && [...dates].map(date => new TravelDateModel(date))
        this._deleted = deleted
        this._edited = edited
        this._travel = new Travel(travel)
        this._createdAt = moment(createdAt).utc(false)
        this._updatedAt = moment(updatedAt).utc(false)
    }

    get v() {
        return this.__v
    }

    get id() {
        return this._id
    }

    get dates() {
        return this._dates
    }

    get deleted() {
        return this._deleted
    }

    get edited() {
        return this._edited
    }

    get travel() {
        return this._travel
    }

    get createdAt() {
        return this._createdAt
    }

    get updatedAt() {
        return this._updatedAt
    }

    hasServices() {
        return this.dates.some(date => date.services.length)
    }

    getRangeDates() {
        if (isNotEmpty(this._travel)) {
            return { endDate: this._travel.travelEndDate, startDate: this._travel.travelStartDate }
        }

        return {}
    }

    getClientsId() {
        return [...this.travel.clients].map(({ _id }) => _id)
    }

    toForm() {
        return {
            dates: this.dates,
            edited: this.edited,
            id: this.id,
            travel: this.travel,
            v: this.v
        }
    }
}

export default TravelScriptModel
