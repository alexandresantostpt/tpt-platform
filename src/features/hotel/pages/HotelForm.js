import { config } from '@config'

import { events } from '@constants/events'
import { librariesServices, services } from '@constants/services'

import PropTypes from 'prop-types'
import PubSub from 'pubsub-js'
import React, { useEffect, useRef, useState } from 'react'

import { colors } from '@helpers/colors'
import { getQueryParams } from '@utils/browser'
import { i18n } from '@i18n'
import { isNotUndefined } from '@utils/functions'
import { withRouter } from 'react-router-dom'

import Button from '@components/button/Button'
import Cancel from '@components/cancel/Cancel'
import FormBuilder from '@components/form/FormBuilder'
import Panel from '@components/panel/Panel'

import FormButtonContainer from '@components/form/container/button/FormButtonContainer'
import WrapperPadding from '@components/wrapper/WrapperPadding'

import { hotelFormFields } from '../fields'

import HotelServiceModel from '../models/HotelServiceModel'

const HotelForm = props => {
    const { match, hotel, hotelSuggestions, resetSuggestionsAction, searchLibraryAction, travelScript } = props
    const {
        params: { id, idService }
    } = match
    const { cityId, dateId, scriptDate, scriptId } = getQueryParams()

    const form = useRef()
    const panel = useRef()
    const previousPath = `/travel/${id}/script`

    const [image, updateImage] = useState()
    const [serviceDisabled, updateServiceDisabled] = useState(false)
    const [newImage, updateNewImage] = useState(null)

    useEffect(() => {
        PubSub.subscribe(events.UPDATE_IMAGE, (_, libraryImage) => {
            updateNewImage(`${config.api.url}/hotel/download/${libraryImage}`)
        })
        return () => PubSub.unsubscribe(events.UPDATE_IMAGE)
    }, [])

    useEffect(() => {
        const { requestGetSingleAction } = props
        if (idService) {
            requestGetSingleAction(idService)
            updateServiceDisabled(true)
        } else {
            requestGetSingleAction(null)
            updateServiceDisabled(false)
        }
    }, [idService])

    useEffect(() => {
        resetSuggestionsAction()
    }, [scriptDate])

    const callSaveData = values => {
        const { requestSaveDataAction } = props

        requestSaveDataAction({
            data: {
                dateId,
                scriptId,
                service: new HotelServiceModel({ ...values, scriptDate }, true).toDTO(),
                type: services.HOTEL
            },
            image,
            redirectTo: previousPath
        })
        reset()
    }

    const callSearchLibrary = name => {
        if (isNotUndefined(cityId)) {
            searchLibraryAction({
                cityId: cityId,
                libraryName: name,
                type: librariesServices.HOTEL,
                updateRecords: true
            })
        }
    }

    const onSaveClick = () => form.current.form.submit()

    const reset = () => {
        PubSub.publish(events.FORM_RESET)
        updateImage(null)
    }

    return (
        <>
            <WrapperPadding>
                <Panel
                    backgroundImageUrl={hotel ? hotel.getImage() : newImage || null}
                    icon="hotel"
                    ref={panel}
                    uploadImage={updateImage}
                />
            </WrapperPadding>
            <FormBuilder
                clients={props.clients}
                disabled={serviceDisabled}
                fields={hotelFormFields}
                getSuggestions={callSearchLibrary}
                initialValues={hotel ? hotel.toForm() : { checkInDate: scriptDate }}
                onSubmit={callSaveData}
                rangeDates={travelScript && travelScript.getRangeDates()}
                ref={process.env.__TEST__ ? null : form}
                resetSuggestions={resetSuggestionsAction}
                suggestions={hotelSuggestions}
                type={services.HOTEL}
                updatePanelImage={true}
            />
            <FormButtonContainer>
                <Button
                    color={colors.purple}
                    onClick={() => (serviceDisabled ? updateServiceDisabled(false) : onSaveClick())}
                    text={i18n.t(serviceDisabled ? 'buttons.editService' : 'buttons.saveService')}
                />
                <Cancel link={previousPath} />
            </FormButtonContainer>
        </>
    )
}

HotelForm.defaultProps = {
    hotel: {},
    hotelSuggestions: [],
    travelScript: {}
}

HotelForm.propTypes = {
    hotel: PropTypes.objectOf(HotelServiceModel),
    hotelSuggestions: PropTypes.array,
    match: PropTypes.object.isRequired,
    requestDelAction: PropTypes.func.isRequired,
    requestGetSingleAction: PropTypes.func.isRequired,
    resetSuggestionsAction: PropTypes.func.isRequired,
    searchLibraryAction: PropTypes.func.isRequired,
    travelScript: PropTypes.object
}

export default withRouter(HotelForm)
