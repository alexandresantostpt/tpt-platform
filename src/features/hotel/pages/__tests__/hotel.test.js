import React from 'react'
import renderer from 'react-test-renderer'

import HotelForm from '../HotelForm'
import HotelServiceModel from '@/features/hotel/models/HotelServiceModel'
import TravelScriptModel from '@/features/travelScript/models/TravelScriptModel'

describe('Tests for Hotel component', () => {
    const hotel = new HotelServiceModel({
        __v: 0,
        _id: '5c6c16bc793e3b4f622f7a9e',
        address: 'Rua endereço',
        checkInDate: '2019-03-01T00:00:00.000Z',
        checkInHour: '2019-03-01T19:00:00.000Z',
        checkOutDate: '2019-03-02T00:00:00.000Z',
        checkOutHour: '2019-03-02T19:00:00.000Z',
        createdAt: '2019-02-21T00:00:00.000Z',
        differences: 'Restarante Michelin no hotel',
        guestCount: '4',
        includedMeals: ['Café da manhã', 'Jantar'],
        name: 'Four seasons',
        roomType: 'Deluxe',
        script: '5c792717e6e603726db85733',
        scriptDate: '2019-03-01T17:10:10.000Z',
        updatedAt: '2019-02-21T00:00:00.000Z'
    })
    const travelScript = new TravelScriptModel({ travel: { __v: 0 } })

    it('Should render withou crash', () => {
        const match = { params: { id: '5c702543793e3b6d3d2f7aa6' } }
        const tree = renderer
            .create(
                <HotelForm
                    hotel={hotel}
                    match={match}
                    requestDelAction={jest.fn()}
                    requestGetSingleAction={jest.fn()}
                    resetSuggestionsAction={jest.fn()}
                    travelRequestGetSingleAction={jest.fn()}
                    travelScript={travelScript}
                />
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
