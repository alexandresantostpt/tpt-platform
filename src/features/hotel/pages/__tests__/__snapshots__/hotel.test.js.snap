// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Hotel component Should render withou crash 1`] = `
Array [
  <div
    className="WrapperPadding-ovjbyx-0 lcnoja"
  >
    <div
      className="Panel__StyledPanel-sc-1qzkpb6-0 cGUlCn"
    >
      <span
        className="Panel__StyledIcon-sc-1qzkpb6-2 iavvSk"
      >
        <span
          className="icon icon-hotel Icon__StyledIcon-crr1cg-0 dfhcZR"
          disabled={false}
          fontSize="3rem"
          onClick={null}
        />
      </span>
      <div
        className="UploadButton__StyledContainer-sc-1p8zoc-0 fpZFIG"
      >
        <label
          className="UploadButton__StyledUploadButton-sc-1p8zoc-3 ivhrYW"
          htmlFor="inputFile"
        >
          +
        </label>
        <input
          accept="image/png, image/jpeg"
          className="UploadButton__InputFile-sc-1p8zoc-4 eLaooA"
          id="inputFile"
          onChange={[Function]}
          type="file"
        />
      </div>
    </div>
    <ul
      className="PanelActions__StyledActions-sc-8xk8c8-0 lllkGw"
    />
  </div>,
  <form
    className="FormBuilder__StyledForm-c1htou-0 PZxQa"
    onSubmit={[Function]}
  >
    <div
      className="form__container"
    >
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Nome do Hotel
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormSuggest__StyledAutosuggest-d8stcq-0 gUlkuf"
        >
          <div
            aria-expanded={false}
            aria-haspopup="listbox"
            aria-owns="react-autowhatever-library.name"
            className="react-autosuggest__container"
            role="combobox"
          >
            <input
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls="react-autowhatever-library.name"
              autoComplete="off"
              className="react-autosuggest__input"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
            <div
              className="react-autosuggest__suggestions-container"
              id="react-autowhatever-library.name"
              role="listbox"
            />
          </div>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Endereço
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-library.address"
          disabled={false}
          id="address"
          maxLength={255}
          name="library.address"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Data de check-in
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormDate__StyledFormDate-sc-1cgz0rk-0 ebKpCI"
        >
          <div
            className="SingleDatePicker SingleDatePicker_1"
          >
            <div>
              <div
                className="SingleDatePickerInput SingleDatePickerInput_1"
              >
                <div
                  className="DateInput DateInput_1"
                >
                  <input
                    aria-describedby="DateInput__screen-reader-message-"
                    aria-label="Selecione uma data..."
                    autoComplete="off"
                    className="DateInput_input DateInput_input_1 DateInput_input__readOnly DateInput_input__readOnly_2"
                    disabled={false}
                    id=""
                    name=""
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Selecione uma data..."
                    readOnly={true}
                    required={false}
                    type="text"
                    value="01 Mar 2019"
                  />
                  <p
                    className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                    id="DateInput__screen-reader-message-"
                  >
                    Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                  </p>
                </div>
                <button
                  className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="icon icon-calendar Icon__StyledIcon-crr1cg-0 cmjVJQ"
                    disabled={false}
                    fontSize="1.75rem"
                    onClick={null}
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Horário de check-in
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormTime__StyledFormTime-sc-1ka3ing-1 bdjJPj"
        >
          <input
            className="FormTime__StyledDatePicker-sc-1ka3ing-0 hCpkqS"
            disabled={false}
            onChange={[Function]}
            options={
              Object {
                "blocks": Array [
                  2,
                  2,
                ],
                "delimiters": Array [
                  ":",
                ],
              }
            }
            placeholderText="Selecione um horário..."
            value="19:00"
          />
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Data de check-out
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormDate__StyledFormDate-sc-1cgz0rk-0 ebKpCI"
        >
          <div
            className="SingleDatePicker SingleDatePicker_1"
          >
            <div>
              <div
                className="SingleDatePickerInput SingleDatePickerInput_1"
              >
                <div
                  className="DateInput DateInput_1"
                >
                  <input
                    aria-describedby="DateInput__screen-reader-message-"
                    aria-label="Selecione uma data..."
                    autoComplete="off"
                    className="DateInput_input DateInput_input_1 DateInput_input__readOnly DateInput_input__readOnly_2"
                    disabled={false}
                    id=""
                    name=""
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Selecione uma data..."
                    readOnly={true}
                    required={false}
                    type="text"
                    value="02 Mar 2019"
                  />
                  <p
                    className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                    id="DateInput__screen-reader-message-"
                  >
                    Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                  </p>
                </div>
                <button
                  className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="icon icon-calendar Icon__StyledIcon-crr1cg-0 cmjVJQ"
                    disabled={false}
                    fontSize="1.75rem"
                    onClick={null}
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Horário de check-out
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormTime__StyledFormTime-sc-1ka3ing-1 bdjJPj"
        >
          <input
            className="FormTime__StyledDatePicker-sc-1ka3ing-0 hCpkqS"
            disabled={false}
            onChange={[Function]}
            options={
              Object {
                "blocks": Array [
                  2,
                  2,
                ],
                "delimiters": Array [
                  ":",
                ],
              }
            }
            placeholderText="Selecione um horário..."
            value="19:00"
          />
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Quantidade de hóspedes
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-guestCount"
          disabled={false}
          id="guestCount"
          maxLength={255}
          min={1}
          name="guestCount"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="number"
          value="4"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 isOWlT"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Código da reserva
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-reserveCode"
          disabled={false}
          id="reserveCode"
          maxLength={15}
          name="reserveCode"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Tipo de quarto
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-roomType"
          disabled={false}
          id="roomType"
          maxLength={120}
          name="roomType"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={true}
          type="text"
          value="Deluxe"
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Refeições incluídas
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
        <div
          className="FormCheck__StyledOption-sc-4ngozm-1 Tgnbj"
        >
          <input
            checked={false}
            id="breakfast"
            name="breakfast"
            onChange={[Function]}
            type="checkbox"
            value="breakfast"
          />
          <span
            className="FormCheck__StyledLabel-sc-4ngozm-0 ijqNlP"
          >
            Café da manhã
          </span>
        </div>
        <div
          className="FormCheck__StyledOption-sc-4ngozm-1 Tgnbj"
        >
          <input
            checked={false}
            id="lunch"
            name="lunch"
            onChange={[Function]}
            type="checkbox"
            value="lunch"
          />
          <span
            className="FormCheck__StyledLabel-sc-4ngozm-0 ijqNlP"
          >
            Almoço
          </span>
        </div>
        <div
          className="FormCheck__StyledOption-sc-4ngozm-1 Tgnbj"
        >
          <input
            checked={false}
            id="dinner"
            name="dinner"
            onChange={[Function]}
            type="checkbox"
            value="dinner"
          />
          <span
            className="FormCheck__StyledLabel-sc-4ngozm-0 ijqNlP"
          >
            Jantar
          </span>
        </div>
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Diferenciais
          
        </label>
        <input
          className="FormField__Field-sc-1at7ilr-0 ceLHeu FormInput__StyledFormInput-sc-77s1r2-0 XtSyS"
          data-cy="input-library.differences"
          disabled={false}
          id="differences"
          maxLength={255}
          name="library.differences"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder=""
          readOnly={false}
          required={false}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormGroup__StyledFormGroup-hydimu-0 vziFA"
      >
        <label
          className="FormLabel__StyledFormLabel-vu10xa-0 cVRKKv"
        >
          Selecionar passageiros
          
          <span
            className="FormRequired__StyledFormRequired-svb7d8-0 jawWZC"
          >
             *
          </span>
        </label>
      </div>
    </div>
  </form>,
  <div
    className="FormButtonContainer-sc-1cqqivn-0 ADyLt"
  >
    <button
      className="Button__StyledButton-sc-1fc3klx-0 bEGBgE"
      disabled={false}
      fill={false}
      onClick={[Function]}
    >
      Salvar serviço
    </button>
    <a
      className="Cancel__StyledCancel-sc-1prgtit-0 WzuoR"
      to="/travel/5c702543793e3b6d3d2f7aa6/script"
    >
      Cancelar
    </a>
  </div>,
]
`;
